provider/pkg/await/deployment.go

```diff
@@ -858,3 +886,35 @@ func (dia *deploymentInitAwaiter) makeClients() (
 
 	return
 }
+
+func (dia *deploymentInitAwaiter) makeInformer(gvr schema.GroupVersionResource, informChan chan<- watch.Event) cache.SharedIndexInformer {
+	informer := dia.config.informerFactory.ForResource(gvr).Informer()
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			informChan <- watch.Event{
+				Object: obj.(*unstructured.Unstructured),
```

Comment:
We should probably do a defensive type assertion here to avoid panics in case the object is the wrong type somehow.